src/extension.ts

```diff
@@ -13,6 +13,26 @@ import { addTagQuickpick, sigmaCompile, onEnterKey, lookup, related, openSigconv
 
 export var attackTags = require("./techniques.json")
 
+function isSigma(doc: vscode.TextDocument) {
+    let isTitle = false;
+    let isDetection = false;
+    let isLogsource = false;
+    for (let i = 0; i < doc.lineCount; i++) {
+      const line = doc.lineAt(i);
+      if (line.text.match(/^title: .*$/)) {
+        isTitle = true;
+      } else if (line.text.match(/^detection: .*$/)) {
+        isDetection = true;
+      } else if (line.text.match(/^logsource: .*$/)) {
+        isLogsource = true;
+      } 
+      if (isTitle && isDetection && isLogsource) {
+        return true;
+      }
+    }
+    return false;
+}
+
```

Comment:
```suggestion
function isSigma(doc: vscode.TextDocument): boolean {
  if (!/^title:\s+.*$/.test(doc.lineAt(0).text)) {
    return false;
  }
  let hasDetection = false;
  let hasLogsource = false;
  const lineCount = Math.min(doc.lineCount, 100);
  for (let i = 1; i < lineCount; i++) {
    const text = doc.lineAt(i).text;
    if (/^detection:\s+.*$/.test(text)) {
      hasDetection = true;
    } else if (/^logsource:\s+.*$/.test(text)) {
      hasLogsource = true;
    }
    if (hasDetection && hasLogsource) {
      return true;
    }
  }
  return false;
}
```

Return false if first line does not start with `title:`
Only check the first 100 lines in case the document is longer this should cover even bigger rules and avoid parsing every line of bigger documents

edit: removed redundant check in the if statements